src/com/dabomstew/pkrandom/romhandlers/Gen5RomHandler.java

```diff
@@ -3461,4 +3462,75 @@ public BufferedImage getMascotImage() {
+        }
+        int[] pokeMoves = RomFunctions.getMovesAtLevel(tp.pokemon.number, movesets, tp.level);
+        for (int moveIdx : pokeMoves) {
+            Move move = moves.get(moveIdx);
+            if (move == null) {
+                continue;
+            }
+            if (move.category == MoveCategory.PHYSICAL) {
+                items.add(Gen4Constants.liechiBerry);
+                items.add(Gen5Constants.consumableTypeBoostingItems.get(move.type));
+                if (!consumableOnly) {
+                    items.addAll(Gen5Constants.typeBoostingItems.get(move.type));
+                    items.add(Gen4Constants.choiceBand);
+                    items.add(Gen4Constants.muscleBand);
+                }
+            }
+            if (move.category == MoveCategory.SPECIAL) {
+                items.add(Gen4Constants.petayaBerry);
+                items.add(Gen5Constants.consumableTypeBoostingItems.get(move.type));
+                if (!consumableOnly) {
+                    items.addAll(Gen5Constants.typeBoostingItems.get(move.type));
+                    items.add(Gen4Constants.wiseGlasses);
+                    items.add(Gen4Constants.choiceSpecs);
+                }
+            }
+            if (!consumableOnly && Gen5Constants.moveBoostingItems.containsKey(moveIdx)) {
+                items.addAll(Gen5Constants.moveBoostingItems.get(moveIdx));
+            }
+        }
+        Map<Type, Effectiveness> byType = Effectiveness.against(tp.pokemon.primaryType, tp.pokemon.secondaryType, 5);
+        for(Map.Entry<Type, Effectiveness> entry : byType.entrySet()) {
+            Integer berry = Gen5Constants.weaknessReducingBerries.get(entry.getKey());
+            if (entry.getValue() == Effectiveness.DOUBLE) {
+                items.add(berry);
+            } else if (entry.getValue() == Effectiveness.QUADRUPLE) {
+                for (int i = 0; i < 8; i ++) {  // Increase the likelihood of protecting against 4x weak.
+                    items.add(berry);
+                }
+            }
+        }
+        if (byType.get(Type.GROUND) != Effectiveness.ZERO && byType.get(Type.GROUND) != Effectiveness.QUARTER) {
+            items.add(Gen5Constants.airBalloon);
+        }
+
+        if (!consumableOnly) {
+            if (tp.ability == Gen4Constants.guts) {
+                items.add(Gen4Constants.flameOrb);
+                items.add(Gen4Constants.toxicOrb);
+            }
+            if (tp.pokemon.primaryType == Type.POISON || tp.pokemon.secondaryType == Type.POISON) {
+                items.add(Gen4Constants.blackSludge);
+            }
+            List<Integer> speciesItems = Gen5Constants.speciesBoostingItems.get(tp.pokemon.number);
+            if (speciesItems != null) {
+                for (int i = 0 ; i < 8; i ++) {  // Increase the likelihood of using species specific items.
+                    items.addAll(speciesItems);
+                }
+            }
+            if (!tp.pokemon.evolutionsFrom.isEmpty()) {

```

Comment:
Both you and Ajarmar can disagree with me here, but I feel like earlygame Pokemon having Eviolite is just super annoying. Maybe gating this behind a level (25? 30? 35?) would make more sense. Again, just sort of spitballing here, feel free to say "no"

Reply:
I'd have to agree - Eviolite stands out as a _very_ strong item, especially very early in the game. That said, since it's rare for non-evolved Pokemon to appear late in the game, it shouldn't have too high of a level gate. I think 25 seems reasonable.

Reply:
I added a level cap of 20 (same as Shedninja, right?) I ran the randomizer for Black giving sensible items to every trainer pokemon. Out of the 1304 trainer pokemon, 6 of them were given Eviolite. 

I then ran it again with "Ironmon" difficulty settings (+50% foe level, fully evolved at lvl 30) and eviolite showed up 2 times.

I tried it again with the limit at level 25. Without modifying trainer level or evolutions, eviolite showed up 6 times again. With "Ironmon" difficulty settings, it showed up 0 times.

To me, level 20 feels fine, given how infrequently it shows up on average. What do y'all think with the above frequencies I outlined?

Reply:
Yes, that's the same level used for "No Early Wonder Guard". I'd be interested in seeing what the probability is *for unevolved Pokemon*. So in your example, 6/1304 trainer Pokemon got it, but how many of those 1304 Pokemon were eligible to get it in the first place? But overall, this probability seems fine to me.

Reply:
Ran it again for Pokemon Black with level cap 20
 - normal levels, no early evolution: 514 out of 1304 were eligible for eviolite; 4 were given it (~1%).
 - "ironmon" difficulty: 51 out of 1304 were eligible for eviolite; 1 was given it (~2%)

Reply:
Yeah those seem like acceptable odds to me. At the very least, it's clear that level 30 is way too high, so if these odds are too high, then we can just bump it to 25.